### PR TITLE
Chore/setup cicd

### PR DESCRIPTION
**Pull request by mosharaf13**
https://github.com/mosharaf13/google-scrapper-ruby/issues/3

## What happened 👀

- Set docker registry host on repo
- Set docker registry token on repo
- Set heroku api token on repo
- Change config DOCKER_REGISTRY_USERNAME and DOCKER_IMAGE to match with docker account

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -98,3 +98,5 @@ end
 group :production do
   gem 'rack-timeout' # Rack middleware which aborts requests that have been running for longer than a specified timeout.
 end
+
+gem "dockerfile-rails", ">= 1.2", :group => :development
```
Comment (sanG-github) [inline Gemfile:102]: I'm curious why we need this gem @mosharaf13? 🤔 

Moreover, if we need to add a gem to the `development` group only, you can add it in the block after line 33. We can do the same for `test`, `production` based on the corresponding group. 😉 

```ruby
group :development do

end
```

File: bin/start.sh [hunk 2]
```diff
@@ -9,4 +9,4 @@ fi
 
 # Run pending migrations (if any) and start rails
 bundle exec rails db:migrate
-bundle exec rails s -p $PORT -b 0.0.0.0
+bundle exec rails s -p 3000 -b 0.0.0.0
```
Comment (sanG-github) [inline bin/start.sh:12]: Should we keep `$PORT` as previously, we can change/manage it easier later 👋

File: config/environments/development.rb [hunk 3]
```diff
@@ -85,4 +85,6 @@
     Bullet.rails_logger  = true
     Bullet.add_footer    = true
   end
+
+  config.hosts << ENV['DEPLOYED_HOST']
```
Comment (sanG-github) [inline config/environments/development.rb:89]: We just need to keep this line in the `production` file, as we just deploy it in the production environment only. 🚀 

```suggestion
```

File: fly.toml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    type = "http"
+    interval = "10s"
+    timeout = "2s"
+    grace_period = "5s"
+    method = "GET"
+    path = "/health_check"
+    protocol = "http"
+
+[env]
+    DEPLOYED_HOST = "google-scrapper-ruby.fly.dev"
```
Comment (sanG-github) [inline fly.toml:29]: It must be 2 spaces indent, no? 🙈 
```suggestion
  DEPLOYED_HOST = "google-scrapper-ruby.fly.dev"
```

**Review by sanG-github — COMMENTED** [review]
I have some questions, the rest looks good to me 🚀

**Review by sanG-github — COMMENTED** [review]
I have some questions, the rest looks good to me 🚀

File: Gemfile (hunk 1 above)
Comment (mosharaf13) [reply Gemfile:102]: It got added when I initialized fly [here](https://github.com/mosharaf13/google-scrapper-ruby/pull/27/commits/2e13f50abd597fef3b429a46bc6ff2a76a4731fc). Maybe it's required by fly to complete cd. Should it be moved inside development group?

File: config/environments/development.rb (hunk 3 above)
Comment (mosharaf13) [reply config/environments/development.rb:89]: Got it 🙌🏼 . Changing ..

**Pull request merged**